src/org/zaproxy/zap/eventBus/SimpleEventBus.java

```diff
@@ -127,31 +128,54 @@ public void registerConsumer(EventConsumer consumer, String publisherName,
 	}
 
 	@Override
-	public void unregisterConsumer(EventConsumer consumer, String publisherName) {
+	public void unregisterConsumer(EventConsumer consumer) {
 		if (consumer == null) {
 			throw new InvalidParameterException("Consumer must not be null");
 		}
 
 		regMgmtLock.lock();
 		try {
-			log.debug("unregisterConsumer " + consumer.getClass().getCanonicalName() + " for " + publisherName);
-			RegisteredPublisher publisher = this.nameToPublisher.get(publisherName);
-			if (publisher != null) {
-				publisher.removeComsumer(consumer);
-			} else {
-				// Check to see if its cached waiting for the publisher
-				for (RegisteredConsumer regCon : this.danglingConsumers) {
-					if (regCon.getConsumer().equals(consumer)) {
-						this.danglingConsumers.remove(regCon);
-						break;
-					}
+			log.debug("unregisterConsumer " + consumer.getClass().getCanonicalName());
+			for (Entry<String, RegisteredPublisher> entry : nameToPublisher.entrySet()) {
+			    entry.getValue().removeComsumer(consumer);
+			}
+			// Check to see if its cached waiting for a publisher
+			for (RegisteredConsumer regCon : this.danglingConsumers) {
+				if (regCon.getConsumer().equals(consumer)) {
+					this.danglingConsumers.remove(regCon);
+					break;
```

Comment:
There can only be one? (below too)

Reply:
That was copied from the existing code :) I guess in theory a consumer could get added more than once, but it def wouldnt be good practice as they'd then get multiple events. @thc202 views?

Reply:
Yeah, that sounds like a bug, we could throw when the same consumer is registered more than once for the same publisher?